File: src/models/usage.ts
export interface UsageStat {
  id: string;
  workspace_id: string;
  chat_id: string;
  message_id: string;
  provider: string;
  model: string;
  input_tokens: number;
  output_tokens: number;
  total_tokens: number;
  latency_ms: number;
  cost: number;
  timestamp: number;
  is_stream: boolean;
  status: string;
  request_type: string;
}

export interface UsageFilter {
  workspace_id?: string;
  provider?: string;
  model?: string;
  date_from?: number;
  date_to?: number;
}

export interface UsageSummary {
  total_input_tokens: number;
  total_output_tokens: number;
  total_cost: number;
  total_requests: number;
  average_latency: number;
}

export interface UsageChartPoint {
  timestamp: string;
  requests: number;
  cost: number;
  tokens: number;
}

File: src-tauri/src/models/usage.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct UsageStat {
    pub id: String,
    pub workspace_id: String,
    pub chat_id: String,
    pub message_id: String,
    pub provider: String,
    pub model: String,
    pub input_tokens: u32,
    pub output_tokens: u32,
    pub total_tokens: u32,
    pub latency_ms: u64,
    pub cost: f64,
    pub timestamp: i64,
    pub is_stream: bool,
    pub status: String,
    pub request_type: String, // "chat", "embedding", etc.
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct UsageFilter {
    pub start_date: Option<i64>,
    pub end_date: Option<i64>,
    pub workspace_id: Option<String>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct UsageSummary {
    pub total_requests: u32,
    pub total_input_tokens: u32,
    pub total_output_tokens: u32,
    pub total_cost: f64,
    pub average_latency: f64,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct UsageChartPoint {
    pub timestamp: i64,
    pub requests: u32,
    pub input_tokens: u32,
    pub output_tokens: u32,
    pub cost: f64,
}
